### PR TITLE
Send user mails asynchronously

### PR DESCRIPTION
**Pull request by phyrog**
We need to override the `send_devise_notification` method so that devise uses active job to send the mails asynchronously.

### CODE REVIEW

File: spec/mail_helper.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+unless defined?(Devise::Mailer)
+  module Devise
+    class Mailer
+      def self.test_message(_)
+        OpenStruct.new(deliver_later: :ok)
+      end
+    end
+  end
+end
```
Comment (eugenk) [inline spec/mail_helper.rb]: Please put this file into `spec/support/`.

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -1,6 +1,7 @@
 # frozen_string_literal: true
 
 require 'rails_helper'
+require 'mail_helper'
```
Comment (eugenk) [inline spec/models/user_spec.rb]: No need for this line because`spec/support/**/*.rb` gets required.

File: spec/mail_helper.rb (hunk 1 above)
Comment (eugenk) [inline spec/mail_helper.rb]: I take back this comment. Please delete this file entirely. This is supposed to be stubbing a method, but as soon as this is loaded, the stub is active for every test. Instead, you should stub this in a `before` block in the spec file:
```ruby
before do
  allow(Devise::Mailer).
    to receive(:test_message).
    and_return(OpenStruct.new(deliver_later: :ok))
end
```

**Comment by phyrog** [comment]
Rebased to resolve conflicts

**Pull request merged**